### PR TITLE
Add Noteblocks and Jukeboxes

### PR DESCRIPTION
**Pull request by PaulBGD**
Adds Noteblocks and Jukeboxes. No Glowkit PR required.

---

_Edited by turt2live_

**Related Links:**
- Ticket: #325 
- Waiting for #442 (or similar)

### CODE REVIEW

**Comment by turt2live** [comment]
win

**Comment by turt2live** [comment]
@PaulBGD A few issues noted with the PR. Considering the size of the PR, these do not seem out of scope:
- Notes should be displayed when changing the note
- Jukeboxes do not stop or drop their disk when broken
- Music does not stop when the disk is ejected

**Comment by PaulBGD** [comment]
- SpaceManiac said to wait, as we don't have that effect coded yet
- You said to set the data to stop it, and so I did
- ^

**Comment by turt2live** [comment]
I said to try it, not to go with it. It should work from the sources I looked at, but if it doesn't here then some kind of debugging needs to be done so that it works. 

As for the effects, understandable. I missed that conversation.

**Comment by turt2live** [comment]
On hold for https://github.com/GlowstoneMC/Glowstone/pull/442

`PRHOLD: Block break in type`

**Comment by dequis** [comment]
Required PR merged!

**Comment by PaulBGD** [comment]
oh hi. I'll get the stuff finished in a bit, just very busy.

**Comment by gdude2002** [comment]
Doesn't build. Bad squishy! http://bamboo.gserv.me/browse/GLOW-SRV-PR444-125

```
18-Nov-2014 07:27:20    /home/bamboo/bamboo-agent-home/xml-data/build-dir/GLOW-SRV-PR444/src/main/java/net/glowstone/entity/GlowPlayer.java:1520: error: cannot find symbol
18-Nov-2014 07:27:20            afterBlockChanges.add(new UpdateBlockEntityMessage(location.getBlockX(), location.getBlockY(), location.getBlockZ(), type.getValue(), nbt));
18-Nov-2014 07:27:20                                      ^
18-Nov-2014 07:27:20      symbol:   class UpdateBlockEntityMessage
18-Nov-2014 07:27:20      location: class GlowPlayer
18-Nov-2014 07:27:20    /home/bamboo/bamboo-agent-home/xml-data/build-dir/GLOW-SRV-PR444/src/main/java/net/glowstone/entity/GlowPlayer.java:1561: error: cannot find symbol
18-Nov-2014 07:27:20            PlayerAchievementAwardedEvent event = new PlayerAchievementAwardedEvent(this, achievement);
18-Nov-2014 07:27:20            ^
18-Nov-2014 07:27:20      symbol:   class PlayerAchievementAwardedEvent
18-Nov-2014 07:27:20      location: class GlowPlayer
18-Nov-2014 07:27:20    /home/bamboo/bamboo-agent-home/xml-data/build-dir/GLOW-SRV-PR444/src/main/java/net/glowstone/entity/GlowPlayer.java:1561: error: cannot find symbol
18-Nov-2014 07:27:20            PlayerAchievementAwardedEvent event = new PlayerAchievementAwardedEvent(this, achievement);
18-Nov-2014 07:27:20                                                      ^
18-Nov-2014 07:27:20      symbol:   class PlayerAchievementAwardedEvent
18-Nov-2014 07:27:20      location: class GlowPlayer
18-Nov-2014 07:27:21    /home/bamboo/bamboo-agent-home/xml-data/build-dir/GLOW-SRV-PR444/src/main/java/net/glowstone/block/blocktype/BlockJukebox.java:45: error: method does not override or implement a method from a supertype
18-Nov-2014 07:27:21        @Override
18-Nov-2014 07:27:21        ^
18-Nov-2014 07:27:21    Note: Some input files use or override a deprecated API.
18-Nov-2014 07:27:21    Note: Recompile with -Xlint:deprecation for details.
18-Nov-2014 07:27:21    4 errors
18-Nov-2014 07:27:21    :compileJava FAILED
```

**Comment by PaulBGD** [comment]
It could be I said I'm very busy and haven't had time to update it?

**Comment by dequis** [comment]
Don't worry he spammed that over all the PRs. No hurry.

**Comment by turt2live** [comment]
@PaulBGD As of 22 days ago this was not merge. Please resolve the issue as soon as possible.

**Comment by PaulBGD** [comment]
I apologize, finals and such. I'll see if I can get it done tonight.

**Comment by turt2live** [comment]
@PaulBGD How are finals? I know you said "tonight" 11 days ago but I also know finals can be overwhelming :P

Hopefully your finals went well, and hopefully you don't need to study for the next week or so ;)

**Comment by PaulBGD** [comment]
Baha, forgot and am 1000 miles away from my computer. I'll see if I can remote in now that I have time. So sorry about taking so long.

**Comment by turt2live** [comment]
@PaulBGD When interacting with the noteblocks the sound duration is incorrect (not completed) and the full range of pitches is not accessible (only about 1/4 of the range is used).

**Comment by turt2live** [comment]
@PaulBGD The last recorded activity on this pull request was 15 days ago, with an issue being identified 4 days ago. Please resolve the concern as soon as possible, thanks.

**Comment by PaulBGD** [comment]
As you were going down the PRs, I was getting frightened as you approached mine. Anyhow, I'm trying to find fixes to both of the issues. I'm not sure why MC isn't playing the full sound duration when I'm telling it to.

**Comment by turt2live** [comment]
Well, the fear should have escaped you now that it's been poked :P 

As for the issues, it may be playing a shorter sound due to some weird logic. I would attempt at fixing the reduced possible range of values and hope that fixes the issue (it may be creating an unintentional rapid fire, causing sounds to sound as cut off).

**Comment by turt2live** [comment]
@PaulBGD any update on the changes requested?

**Comment by PaulBGD** [comment]
@turt2live I've been unable to find the cause.

**Comment by mamifsidtect** [comment]
I know it's probably not my place to check, but how is the progression with this issue?

**Comment by PaulBGD** [comment]
@mamifsidtect busy, if anyone would like to try to fix the changes @turt2live talked about then I'd be fine with that.

**Comment by turt2live** [comment]
For the record, if anyone does decide to assist, please do the changes on your own fork and PR them to @PaulBGD's branch, leaving a comment here to indicate as such.

**Comment by turt2live** [comment]
@PaulBGD Any updates on this?

File: src/main/java/net/glowstone/net/handler/play/player/DiggingHandler.java [hunk 1]
```diff
@@ -101,6 +101,9 @@ public void handle(GlowSession session, DiggingMessage message) {
         } else if (revert) {
             // replace the block that wasn't really dug
             BlockPlacementHandler.revert(player, block);
+        } else {
+            BlockType blockType = ItemTable.instance().getBlock(block.getType());
+            blockType.leftClickBlock(player, block, holding);
```
Comment (Tonodus) [inline src/main/java/net/glowstone/net/handler/play/player/DiggingHandler.java:106]: Missing null check?

**Comment by turt2live** [comment]
@PaulBGD This is a final warning: Please update your pull request as soon as possible to have review continue. Thanks!

**Comment by turt2live** [comment]
Closing this as inactive. It's been months without communication and no visible progress. If this is not inactive, please reply as soon as possible @PaulBGD.

If anyone else would like to pick up on this please open a new PR with the appropriate changes. Thank you for your contribution!

**Comment by PaulBGD** [comment]
This is inactive.

**Pull request closed without merging**